docs/_ext/autoclass_with_examples.py

```diff
@@ -0,0 +1,94 @@
+import re
+import sys
+from typing import Any, Dict, List, Tuple
+
+from docutils.nodes import Node
+from sphinx.ext.autodoc.directive import AutodocDirective
+
+from starknet_py.constants import ROOT_PATH
+
+
+class AutoclassWithExamples(AutodocDirective):
+    """
+    Directive class extending AutodocDirective and adding code examples.
+    """
+
+    def run(self) -> List[Node]:
+        original_class = getattr(
+            sys.modules[self.env.ref_context.get("py:module")], self.arguments[0]
+        )
+        add_code_examples(original_class)
+
+        self.name = self.name.replace("-with-examples", "")  # remove `-with-examples`
+
+        return AutodocDirective.run(self)
+
+
+def add_code_examples(original_class: Any):
+    """
+    Adds code example for the given class.
+    """
+    base_class = original_class.__base__
+
+    file_name, file_content = _extract_file_properties(original_class.__name__)
+
+    for method_name, method in original_class.__dict__.items():
+        stripped_method_name = method_name.strip("_")
+        if f"""test_{stripped_method_name}(""" in file_content:
+            hint = create_hint(file_name, stripped_method_name)
+
+            # if method does not have __doc__ take it from the base method
+            if (
+                callable(method)
+                and method.__doc__ is None
+                and method_name in base_class.__dict__
+            ):
+                method.__doc__ = getattr(base_class, method_name).__doc__ + hint
+            elif callable(method):
+                method.__doc__ = (method.__doc__ or "") + hint
```

Comment:
Maybe something like this would be easier to understand
```
if not callable:
   continue
stripped_method_name = method_name.strip("_")
if exists_code_example(stripped_method_name, file_content):
   hint = create_hint
   append_hint(method, hint)
```